docs/products/opensearch/howto/get-started-with-nodejs.rst

```diff
@@ -0,0 +1,528 @@
+
+You probably noticed that we haven’t specified any structure for the documents. Even though we could have set explicit mapping beforehand, we opted to rely on OpenSearch to derive the structure from the data and use dynamic mapping. The derived properties will be sufficient for our examples. Let’s see what properties were defined by OpenSearch when indexing the data.
+
+.. code-block:: javascript
+
+    /**
+     * Retrieving mapping for the index.
+     */
+    module.exports.getMapping = () => {
+        console.log(`Retrieving mapping for ${indexName}`);
+
+        client.indices.getMapping({index: indexName}, (error, result) => {
+            if (error) {
+                console.error(error);
+            } else {
+                console.log(result.body.recipes.mappings.properties);
+            }
+        })
+    };
+
+::
+
+    run-func index.js getMapping
+
+You should be able to see the following structure:
+
+.. code-block:: javascript
+
+    {
+      calories: { type: 'long' },
+      categories: { type: 'text', fields: { keyword: [Object] } },
+      date: { type: 'date' },
+      desc: { type: 'text', fields: { keyword: [Object] } },
+      directions: { type: 'text', fields: { keyword: [Object] } },
+      fat: { type: 'long' },
+      ingredients: { type: 'text', fields: { keyword: [Object] } },
+      protein: { type: 'long' },
+      rating: { type: 'float' },
+      sodium: { type: 'long' },
+      title: { type: 'text', fields: { keyword: [Object] } }
+    }
+
+These are the fields we'll be playing with. You can find information on dynamic mapping types `in the documentation <https://opensearch.org/docs/opensearch/rest-api/create-index/#dynamic-mapping-types>`_.
+
+Searching queries
+-----------------
+
+Now that we have data in the OpenSearch cluster, it is time to run some search queries.
+We can separate possible search queries into three groups: term-level, full-text and boolean. Let's look at each one in more details.
+
+
+Term-level queries
+^^^^^^^^^^^^^^^^^^
+
+Term-level queries are handy when we need to find exact matches for numbers, dates or tags and when we don't need to sort results by relevance.
+
+One of the examples for term-level query is searching for all entries with sodium value equal to 0.
+
+.. code-block:: javascript
+
+    /**
+     * Searching for exact matches of a value in a field.
+     */
+    module.exports.termSearch = (field, value) => {
+        console.log(`Searching for recipes with ${field} equal to ${value}`);
+        const body = {
+            'query': {
+                'term': {
+                    [field]: value
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+::
+
+    run-func index.js termSearch sodium 0
+
```

Comment:
And what? How about teasing people to try it themselves with a specific ingredient? I immediately searched for things with lemon in, to understand how the command worked.